airflow/config_templates/config.yml

```
@@ -116,6 +116,15 @@ core:
       type: string
       example: ~
       default: "16"
+    max_failure_runs_per_dag:
+      description: |
+        The maximum number of consecutive DAG failures before DAG is turned off. The scheduler will disable the DAG
+        if it reaches the limit. This is configurable at the DAG level with ``max_failure_runs``,
+        which is defaulted as ``max_failure_runs_per_dag``.
```

```suggestion
        The maximum number of consecutive failed DAG runs before a DAG is automatically paused. This is also configurable per DAG with the argument ``max_consecutive_failed_dag_runs``.
```

WDYT of having the same name for the global config and per-DAG config for clarity? This avoids config name sprawl.